### PR TITLE
Validate top level ParameterSets like options

### PR DESCRIPTION
**Pull request by wddgit**
One change of behavior comes with this PR. Previously if the configuration did not explicitly specify numberOfStreams it defaulted to 1. After this PR it will default to the number of threads.

### CODE REVIEW

**Comment by wddgit** [comment]
please test

File: FWCore/ParameterSet/interface/validateTopLevelParameterSets.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#ifndef FWCore_ParameterSet_validateTopLevelParameterSets_h
+#define FWCore_ParameterSet_validateTopLevelParameterSets_h
+
+namespace edm {
+
+  class ParameterSet;
+  class ParameterSetDescription;
+
+  const unsigned int s_defaultNumberOfThreads = 1;
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/interface/validateTopLevelParameterSets.h]: How about `constexpr`

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

File: FWCore/ParameterSet/interface/validateTopLevelParameterSets.h (hunk 1 above)
Comment (wddgit) [reply FWCore/ParameterSet/interface/validateTopLevelParameterSets.h]: Done. I pushed a change that makes it constexpr.

**Comment by wddgit** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**